src/librustc_typeck/diagnostics.rs

```diff
@@ -1278,6 +1278,15 @@ fn main() {
 ```
 "##,
 
+E0103: r##"
+You hit this error because the compiler the compiler lacks information
+to determine a type for this pattern binding.
+
+You have two possibilities to solve this situation:
+ * Give an explicit definition of the expression
+ * Infer the expression
```

Comment:
Yeah, but neither suggestion makes sense (or is useful) when talking about patterns.